protocols/autonatv2/src/client/behaviour.rs

```diff
@@ -81,148 +51,240 @@ where
 {
     type ConnectionHandler = Handler;
 
-    type ToSwarm = Event;
+    type ToSwarm = ();
 
     fn handle_established_inbound_connection(
         &mut self,
-        _connection_id: ConnectionId,
-        peer: PeerId,
+        connection_id: ConnectionId,
+        _peer: PeerId,
         _local_addr: &Multiaddr,
-        _remote_addr: &Multiaddr,
-    ) -> Result<Self::ConnectionHandler, ConnectionDenied> {
-        Ok(Handler::new(self.accepted_nonce.clone(), peer))
+        remote_addr: &Multiaddr,
+    ) -> Result<<Self as NetworkBehaviour>::ConnectionHandler, ConnectionDenied> {
+        if addr_is_local(remote_addr) {
+            self.local_peers.insert(connection_id);
+        }
+        Ok(new_handler())
```

Comment:
Note that inbound connections will _always_ be dial-backs and outbound connections will always be dial-requests. Thus I'd suggest to use `Either<dial_back::Handler, dial_request::Handler>` here instead of `ConnectionHandlerSelect` because there is no case in which both of them need to be active. We statically know that we don't want to receive dial-requests on inbound connections for the client.

Reply:
I think we resolved that.